packages/toolpad-app/src/toolpadDataSources/rest/server.ts

```diff
@@ -138,6 +139,21 @@ async function execBase(
     resolveBindableEntries(fetchQuery.headers || [], params),
   ]);
 
+  if (process.env.TOOLPAD_DEMO) {
+    const demoUrls = DEMO_BASE_URLS.map((baseUrl) => baseUrl.url);
+
+    const hasNonDemoConnectionParams =
+      !demoUrls.includes(connection?.baseUrl || '') ||
+      (connection?.headers && connection?.headers.length > 0) ||
+      !!connection?.authentication;
+    const hasNonDemoQueryParams =
+      fetchQuery.method !== 'GET' || (fetchQuery.headers && fetchQuery.headers.length > 0);
+
+    if (hasNonDemoConnectionParams || hasNonDemoQueryParams) {
+      throw new Error(`Cannot use unsupported features in demo version.`);
```

Comment:
This sounds like we do not support these features at all, maybe better wording would be something like `Can not use these features in demo version`?